commit c90c0a8924e65aceb0c1e7b00940db3d8b70fca4 (refs/changes/76/3976/1)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2018-05-18T18:21:16+02:00 (1 year, 5 months ago)
    
    TRITON-397 NAT instances w/o resolvers make `sdcadm fix-core-vm-resolvers` fail

diff --git a/lib/steps/binder.js b/lib/steps/binder.js
index 691b880..519159d 100644
--- a/lib/steps/binder.js
+++ b/lib/steps/binder.js
@@ -226,12 +226,17 @@ function checkCoreVmInstancesResolvers(arg, cb) {
                     func: function checkResolvers(vm, nextVm) {
                         // Binder VMs and eventually, anything we want to
                         // avoid resolvers:
-                        if (vm.resolvers.length === 0 &&
+                        if ((vm.resolvers && vm.resolvers.length === 0) &&
                             vm.internal_metadata &&
                             (vm.internal_metadata.set_resolvers === false)) {
                             nextVm();
                             return;
                         }
+                        // NAT VMs:
+                        if (!vm.resolvers) {
+                            nextVm();
+                            return;
+                        }
 
                         var hasAdminNic = false;
                         var unknownNicTag = false;
